ENH: Add more Lo APIDs

greglucas opened the pull request:
- Add string data type to the excel to xtce converter. It is currently assuming fixed length strings. There is no explicit mention for how to handle the strings in the Excel file and no test data to verify against.
- Add NHK, SHK, and PCC APIDs to the processing code.

This required updating the Excel file delivered to us:
- Only keep the specific packets desired in the Packets tab
- Make SCI_CNT and SCI_DE BYTE fields and get rid of all the individual entries
- find and replace names of variables in the SPIN packet definition. Two fields are named different in the Excel file vs the validation files. `PERIOD_STATE_SPIN_XX` and `PHASE_STATE_SPIN_XX` are renamed to `VALID_PERIOD_SPIN_XX` and `VALID_PHASE_SPIN_XX` respectively.

@sdhoyt, do you have any test / verification files for these apids? It looks like there are nominal housekeeping packets in the packet file but not the static housekeeping or pivot platform apids. The other thing I'm not certain about is whether those "string" versions are truly that long and padded to the full length or if they are dynamic and terminated and that is the max-length. Note that the algorithm document doesn't even have all of the entries for the SHK file in it from what I can tell, so that must have been updated at some point.

closes #1869
closes #1870
closes #1871

greglucas committed:
ENH: Add more Lo APIDs

- Add string data type to the excel to xtce converter. It is currently
  assuming fixed length strings. There is no explicit mention for how
  to handle the strings in the Excel file and no test data to verify against.
- Add NHK, SHK, and PCC APIDs to the processing code.

sdhoyt commented:
I'll reach out to Lo and see if they have sample data they can send me

sdhoyt reviewed (approved):
This all looks great, thank you! I emailed Lo and asked for sample data.

greglucas committed:
Add L1b derived processing to housekeeping datasets

This adds degrees and other engineering units output which is what
is the more desired product. Produce both L1a and L1b at the same time
rather than as separate jobs.

greglucas committed:
Adding test for XTCE string conversion

greglucas committed:
Adding a test for Lo housekeeping l1a/l1b

greglucas commented:
Added more tests:
* NHK l1a
* NHK l1b
* excel_to_xtce string converter

We don't have any SHK packets in this file, so that path is untested right now. Merging over that codecov patch failure.
